src/app/devices/deviceContent/components/deviceIdentity/deviceIdentity.tsx

```diff
@@ -136,6 +150,108 @@ export default class DeviceIdentityInformation
+        const options: IDropdownOption[] = [
+            {
+                key: identity.authentication.symmetricKey.primaryKey,
+                text: context.t(ResourceKeys.deviceIdentity.authenticationType.symmetricKey.primaryKey)
+            },
+            {
+                key: identity.authentication.symmetricKey.secondaryKey,
+                text: context.t(ResourceKeys.deviceIdentity.authenticationType.symmetricKey.secondaryKey)
+            }
+        ];
+
+        const position = Position && Position.top || 0;
+
+        return (
+            <CollapsibleSection
+                expanded={false}
+                label={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.label)}
+                tooltipText={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.toolTip)}
+            >
+                <div className="sas-token-section">
+                    <Dropdown
+                        label={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.symmetricKey)}
+                        selectedKey={sasTokenSelectedKey || undefined}
+                        options={options}
+                        styles={{
+                            dropdown: {
+                                marginBottom: 5,
+                                width: 300
+                            }
+                        }}
+                        onChange={this.onSelectedKeyChanged}
+                    />
+                    <SpinButton
+                        label={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.expiration)}
+                        labelPosition={position}
+                        min={0}
+                        max={Number.MAX_SAFE_INTEGER}
+                        onBlur={this.onExpirationChanged}
+                        styles={{
+                            root: {
+                                marginBottom: 5,
+                                width: 300
+                            }
+                        }}
+                    />
+                    <MaskedCopyableTextField
+                        ariaLabel={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.textField.ariaLabel)}
+                        label={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.textField.label)}
+                        value={this.state.sasTokenConnectionString}
+                        allowMask={true}
+                        t={context.t}
+                        readOnly={true}
+                    />
+                    <PrimaryButton
+                        title={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.generateButton.title)}
+                        text={context.t(ResourceKeys.deviceIdentity.authenticationType.sasToken.generateButton.text)}
+                        onClick={this.onGenerateSASClicked}
+                        disabled={sasTokenSelectedKey === ''}
```

Comment:
If expiration is left blank, are we not disabling this button? Maybe we can give it a five minute default or something accoarding to GenerateSasTokenParameters' description?

Reply:
There should already be a default of 5 minutes in `generateSasToken` if no expiration is provided.